src/placement/jump_map_versions.c

```diff
@@ -304,6 +308,17 @@ __get_target_v1(struct pool_domain *curr_dom, struct pool_target **target,
 						allow_version);
 			if (range_set) {
 				if (top == -1) {
+					if (curr_pd != root_pos) {
+						/* all domains within the PD are full, ignore the
+						 * PD restrict.
+						 */
+						D_DEBUG(DB_PL, "PD[%d] all doms are full, weak the "
+							"PD restrict\n",
+							(int)(curr_dom - root_pos));
+						curr_pd = root_pos;
+						curr_dom = curr_pd;
```

Comment:
hmm, I thought you need re-generate the pd, instead of weak the pd? since there might be other PD yet? or I miss sth?

Btw: I thought __get_target_v1 () can handle multiple domain layer anyway, why separate the pd in a different function, so you may see problem like this?

Reply:
for weak the PD, it is only for the case that there are small number of PD and in each PD with small number of fault doms. There is a test case of "pd_use_all_dom", with 2 PD, 4 fdom in each PD and 1 tgt in each fdom (8 tgts totally), to place OC_RP_4G2 obj with pda 3, without the code here that test case cannot pass. In this case If there are 3 PDs in the system it will not get in this code block.

"I thought __get_target_v1 () can handle multiple domain layer anyway, why separate the pd in a different function, so you may see problem like this"
the current way in this patch seems with less code change, and I tried to avoid adding another bitmap for the PD. we can discuss more details later.